upstream.wrap

```diff
@@ -0,0 +1,6 @@
+[wrap-file]
+directory = eigen-3.3.9
+
+source_url = https://gitlab.com/libeigen/eigen/-/archive/3.3.9/eigen-3.3.9.tar.bz2
+source_filename = eigen-3.3.9.tar.bz2
+source_hash = 0fa5cafe78f66d2b501b43016858070d52ba47bd9b1016b0165a7b8e04675677
```

Comment:
Upstream provides an `eigen3.pc` providing this dependency, usable via `dependency('eigen3')`.

Per https://mesonbuild.com/Wrap-dependency-system-manual.html#provide-section, please include:
```
[provide]
eigen3 = eigen_dep
```

This lets automatic fallback work, and we want "provide" sections for all new wraps.

Reply:
Copy.  Done.  Thanks for the quick and thorough feedback!